tests/sles4sap/forkbomb.pm

```diff
@@ -27,6 +28,23 @@ sub run {
     my $sid = get_required_var('INSTANCE_SID');
     my $instance_id = get_required_var('INSTANCE_ID');
     my $sapadm = $self->set_sap_info($sid, $instance_id);
+
+    my $package_version = script_output "rpm -q --qf '%{VERSION}' saptune";
+    record_info('saptune version', $package_version);
+    record_info('Original kernel.pid_max', script_output('sysctl -a | grep kernel.pid_max', proceed_on_failure => 1));
+    my $note_id = '2578899';
+    record_info("Original conf value", script_output("grep kernel.pid_max /usr/share/saptune/notes/$note_id", proceed_on_failure => 1));
+    if (package_version_cmp($package_version, '3.1.3') >= 0) {
```

Comment:
I'd try to avoid to depend on a specific package version.

Here the problem is more how many process we can support on openQA side.

1. we know that in openQA we have some "HW" limitation and we can at max run the forkbomb test with **N** process ( we know 32768  is fine but it could be even 40000 maybe or something)

2. saptune is in charge to  set pid_max. So the value in the OS should match the one in the saptune configuration. We can maybe test it somehow, or maybe some other test is already in charge of this aspect

3. if the value set by saptune is less than **N**, max number we can support in openQA, we are fine to run the forkbomb test as we run it today. Let saptune to set its value, lets the forkbomb run without arguments and so to create as much process as possible, that should be the number set by saptune itself

4. if the value set by saptune is more than **N**, lets test that at list this large number is effectively set in the OS, then change pid_max  to **N** and run the forkbomb test.

Reply:
My point is that there's a new feature in saptune that for the moment we are not able to test in openQA. Instead of disabling/reverting this feature to let the test to PASs, I propose to get control of what we can test with the openQA resource we know we have (QE to be in control of **N** ).

Reply:
If test case doesn't depend on a specific package version the kernel.pid_max=xxx set in note config file will not take effect on version < 3.1.3 .
For  example 3.1.2, on this version it does not support kernel.pid_max in note configure file.
There is a 3.1.2 VR for this: 
It configured kernel.pid_max=40000 at https://openqa.suse.de/tests/15124658#step/forkbomb/53
and the check the OS actual kernel.pid_max value is 73728 at: https://openqa.suse.de/tests/15124658#step/forkbomb/65

Reply:
How to know the **N** (max number we can support in openQA)?

Reply:
Btw, I opened a new ticket for your suggestions: [TEAM-9587](https://jira.suse.com/browse/TEAM-9587)
enhance forkbomb sub-test for saptune 3.1.3

Reply:
> How to know the **N** (max number we can support in openQA)?

trial and error maybe

Reply:
> I'd try to avoid to depend on a specific package version.
> 
> Here the problem is more how many process we can support on openQA side.
> 
> 1. we know that in openQA we have some "HW" limitation and we can at max run the forkbomb test with **N** process ( we know 32768  is fine but it could be even 40000 maybe or something)
> 2. saptune is in charge to  set pid_max. So the value in the OS should match the one in the saptune configuration. We can maybe test it somehow, or maybe some other test is already in charge of this aspect
> 3. if the value set by saptune is less than **N**, max number we can support in openQA, we are fine to run the forkbomb test as we run it today. Let saptune to set its value, lets the forkbomb run without arguments and so to create as much process as possible, that should be the number set by saptune itself
> 4. if the value set by saptune is more than **N**, lets test that at list this large number is effectively set in the OS, then change pid_max  to **N** and run the forkbomb test.

I did some enhancements (kept the package version check; added OPENQA_KERNEL_PID_MAX  setting; added kernel_pid_max comparison between config file and OS) , 
And did VRs accordingly (still using KERNEL_PID_MAX not OPENQA_KERNEL_PID_MAX in Settings as I did not redo the VRs after renamed it).
Please have a look. 
- 15sp2 (when saptune>=3.1.3) (make test failed on purpose) https://openqa.suse.de/tests/15125031#step/forkbomb/59 (failed as expected) (Error, os_kernel_pid_max_new != conf_kernel_pid_max_new: 40000 != 4194304)
- 15sp2 (when saptune>=3.1.3)(KERNEL_PID_MAX=NULL) https://openqa.suse.de/tests/15125100 (passed as expected)
- 15sp2 (when saptune>=3.1.3)(KERNEL_PID_MAX=5000) https://openqa.suse.de/tests/15125099 (passed as expected)
- 15sp2 (when saptune<3.1.3) (KERNEL_PID_MAX=NULL)https://openqa.suse.de/tests/15125101 (passed as expected)